app/controllers/api/account.php

```diff
@@ -98,6 +99,14 @@
             }
         }
 
+        // Makes sure this email is not already used in another identity
+        $identityWithMatchingEmail = $dbForProject->findOne('identities', [
```

Comment:
Should a user be able to log in with an email associated solely with the provider that is different than the main email?

Are we blocking users to attach OAuth providers with emails associated to other users?

Reply:
> Should a user be able to log in with an email associated solely with the provider that is different than the main email?

Using OAuth2, If the email is tied to an existing user via an identity, they won't be able to log in. That's what this check does:

https://github.com/appwrite/appwrite/blob/8fe688a95253b2b78126b3607fbee4ad60f328be/app/controllers/api/account.php#L564-L570

Using OAuth2, If the email is not associated with an existing identity or user, a new user will be created.

Using Create Email Session, a user must use their main user email and cannot use an identity email.

> Are we blocking users to attach OAuth providers with emails associated to other users?

Yes, here:

https://github.com/appwrite/appwrite/blob/8fe688a95253b2b78126b3607fbee4ad60f328be/app/controllers/api/account.php#L500-L511